cerberus-tables/src/ast/mod.rs

```diff
@@ -349,5 +367,123 @@ pub enum ErrorKind {
     Unexpected(String),
 }
 
+impl std::fmt::Display for ErrorKind {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
 #[cfg(test)]
-mod tests {}
+mod tests {
+    use super::parser;
+    use super::*;
+
+    fn format_errors(errors: Vec<Error>) -> String {
+        let mut msg = String::new();
+        for error in errors {
+            msg += format!("{}", error).as_str();
+        }
+        msg
+    }
+
+    fn assert_row_content(
+        row: &TableRow<Type>,
+        expected_path: &str,
+        expected_name: &str,
+        expected_desc: &str,
+        expected_cursors: (parser::Cursor, parser::Cursor),
+    ) {
+        match &row.value.kind {
+            TypeKind::Path(path) => {
+                assert_eq!(path.span.text(), expected_path);
+            }
+            _ => {
+                assert_eq!(
+                    true, false,
+                    "Incorrect type kind: {:?}",
+                    row.value.kind
+                );
+            }
+        }
+        assert_eq!(row.name.name(), expected_name);
+        assert_eq!(row.desc.unwrap().text(), expected_desc);
+        assert_eq!(row.span.range.0, expected_cursors.0);
+        assert_eq!(row.span.range.1, expected_cursors.1);
+    }
+
+    #[test]
+    fn markdownfile_parse_tables() {
```

Comment:
Something I might recommend for when we add more tests is to do something like rustc's UI tests: https://github.com/rust-lang/rust/blob/master/src/test/ui/fn/fn-recover-return-sign2.rs#L5

The idea being that this makes it easier to find spans instead of having to hand-code them, which can be quite painful. You can pull in files via `include_str!()`.